core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4MultiQueriesPreparedStatement.java

```diff
@@ -0,0 +1,393 @@
+                System.arraycopy(result, 0, tmp, results.length, result.length);
+                results = tmp;
+            }
+        }
+
+        return results;
+    }
+
+    // TODO here may add some logic in TrafT4ResultSet
+    public ResultSet executeQuery() throws SQLException {
+        if (connection_.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection_.props_);
+            connection_.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "executeQuery", "", p);
+        }
+        ResultSet[] results = new ResultSet[pstmtArr.length];
+        currentSqlIndex = 0;
+
+        for (int i = 0; i < pstmtArr.length; i++) {
+            results[i] = pstmtArr[i].executeQuery();
+        }
+        return results[currentSqlIndex++];
+    }
+
+    public boolean execute() throws SQLException {
+        if (connection_.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection_.props_);
+            connection_.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "execute", "", p);
+        }
+        boolean result = false;
+        currentSqlIndex = 0;
+
+        for (int i = 0; i < pstmtArr.length; i++) {
+            result |= pstmtArr[i].execute();
+        }
+        return result;
+    }
+
+    public int getUpdateCount() throws SQLException {
+        if (connection_.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection_.props_);
+            connection_.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "getUpdateCount", "",
+                    p);
+        }
+        long result = 0;
+        currentSqlIndex = 0;
+
+        for (int i = 0; i < pstmtArr.length; i++) {
+            result += pstmtArr[i].getUpdateCount();
+        }
+        return (int) result;
+    }
+
+    public int executeUpdate() throws SQLException {
+        if (connection_.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection_.props_);
+            connection_.props_.t4Logger_
+                    .logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "executeUpdate", "", p);
+        }
+        long result = 0;
+        currentSqlIndex = 0;
+
+        for (int i = 0; i < pstmtArr.length; i++) {
+            result += pstmtArr[i].executeUpdate();
+        }
+        return (int) result;
+    }
+
+    public void close() throws SQLException {
+        if (connection_.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection_.props_);
+            connection_.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "close", "", p);
+        }
+        for (int i = 0; i < pstmtArr.length; i++) {
+            pstmtArr[i].close();
```

Comment:
Would it be helpful to add a try catch around here and close as many statements as possible and throw the first exception, so as to free resources. For example I have 5 statements and 2nd one fails, the remaining 3 also will not get closed.

Reply:
very good suggestions, thanks